[BUG] Fix org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=search/110_field_collapsing/field collapsing and search_after} flaky test

**Describe the bug**
The org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=search/110_field_collapsing/field collapsing and search_after} test case is flaky.

```
java.lang.AssertionError: Failure at [search/110_field_collapsing:254]: expected [4xx|5xx] status code but api [search] returned [200 OK] [{"took":3,"timed_out":false,"_shards":{"total":2,"successful":1,"skipped":1,"failed":1,"failures":[{"shard":0,"index":"test","node":"Y3WKi7AWToWkPSGJCPpURA","reason":{"type":"search_exception","reason":"cannot use `collapse` in conjunction with `search_after`"}}]},"hits":{"total":0,"max_score":0.0,"hits":[]}}]
	at __randomizedtesting.SeedInfo.seed([9A807F988B26D31A:12D4404225DABEE2]:0)
	at org.opensearch.test.rest.yaml.OpenSearchClientYamlSuiteTestCase.executeSection(OpenSearchClientYamlSuiteTestCase.java:460)
	at org.opensearch.test.rest.yaml.OpenSearchClientYamlSuiteTestCase.test(OpenSearchClientYamlSuiteTestCase.java:433)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750
```
**To Reproduce**

```
./gradlew ':qa:mixed-cluster:v2.9.0#mixedClusterTest' --tests "org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=search/110_field_collapsing/field collapsing and search_after}" -Dtests.seed=9A807F988B26D31A
```

**Expected behavior**
Test should always pass

**Plugins**
Standard

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Host/Environment (please complete the following information):**
 - CI

**Additional context**
- https://build.ci.opensearch.org/job/gradle-check/16677/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_search_110_field_collapsing_field_collapsing_and_search_after_/

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/search/110_field_collapsing.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/search/110_field_collapsing.yml
@@ -254,6 +254,7 @@ setup:
   - do:
       catch:      /cannot use \`collapse\` in conjunction with \`search_after\`/
       search:
+        allow_partial_search_results: false
         rest_total_hits_as_int: true
         index: test
         body:
@@ -267,6 +268,7 @@ setup:
   - do:
       catch:      /cannot use \`collapse\` in conjunction with \`rescore\`/
       search:
+        allow_partial_search_results: false
         rest_total_hits_as_int: true
         index: test
         body:

